dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,282 @@
+            }
+        }
+
+        ct.ThrowIfCancellationRequested();
+        throw new KernelException("This code is unreachable.");
+    }
+
+    /// <summary>
+    /// Create a URL string that includes the default BaseUrl if the BaseAddress on _httpClient isn't set.
+    /// </summary>
+    /// <param name="url">URL without base.</param>
+    /// <returns>URL with or without BaseUrl.</returns>
+    private string Url(string url)
+    {
+        return this._httpClient.BaseAddress is null ? $"{BaseUrl}{url}" : url;
+    }
+
+    private void AddDefaultHeaders(HttpRequestMessage request)
+    {
+        request.Headers.Authorization = new AuthenticationHeaderValue(this._apiKey);
+        request.Headers.Add("User-Agent", HttpHeaderConstant.Values.UserAgent);
+        request.Headers.Add(HttpHeaderConstant.Names.SemanticKernelVersion,
+            HttpHeaderConstant.Values.GetAssemblyVersion(typeof(AssemblyAIAudioToTextService)));
+    }
+
+    private async Task<HttpResponseMessage> SendWithSuccessCheckAsync(HttpClient client, HttpRequestMessage request, CancellationToken ct)
+    {
+        HttpResponseMessage? response = null;
+        try
+        {
+            response = await client.SendAsync(request, HttpCompletionOption.ResponseContentRead, ct).ConfigureAwait(false);
+        }
+        catch (HttpRequestException e)
+        {
+            throw new HttpOperationException(HttpStatusCode.BadRequest, null, e.Message, e);
+        }
+
+        if (response.IsSuccessStatusCode)
+        {
+            return response;
+        }
+
+        string? responseContent = null;
+        try
+        {
+            // On .NET Framework, EnsureSuccessStatusCode disposes of the response content;
+            // that was changed years ago in .NET Core, but for .NET Framework it means in order
+            // to read the response content in the case of failure, that has to be
+            // done before calling EnsureSuccessStatusCode.
+            responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+            if (response.Content.Headers.ContentType.MediaType == "application/json")
+            {
+                var json = JsonDocument.Parse(responseContent);
+                if (json.RootElement.TryGetProperty("error", out var errorProperty))
+                {
+                    throw new HttpOperationException(
+                        statusCode: response.StatusCode,
+                        responseContent: responseContent,
+                        message: errorProperty.GetString()!,
+                        innerException: null
+                    );
+                }
+            }
+
+            response.EnsureSuccessStatusCode(); // will always throw
+        }
+        catch (Exception e)
+        {
+            throw new HttpOperationException(response.StatusCode, responseContent, e.Message, e);
+        }
+
+        throw new KernelException("Unreachable code.");
```

Comment:
If the code is unreachable, what is it for?

Reply:
```csharp
            response.EnsureSuccessStatusCode(); // will always throw
        }
        catch (Exception e)
        {
            throw new HttpOperationException(response.StatusCode, responseContent, e.Message, e);
        }
```
Because the HTTP response at this point is not a success code, `EnsureSuccessStatusCode()` will always throw.
Inside of the catch, it will always throw.
So `throw new KernelException("Unreachable code.")` is unreachable, but for the compiler it needs to return or throw something.

Reply:
If you know `ensure` method will always throws in your case, just throw new right exception. This will be more elegant :)